### PR TITLE
GLOQC: Reduce memory footprint

### PR DESCRIPTION
**Pull request by f3sch**
- stop creation of MC histos in Data

### CODE REVIEW

**Comment by f3sch** [comment]
Changing to ready to test CI.

File: Detectors/GlobalTracking/include/GlobalTracking/MatchITSTPCQC.h
Comment (chiarazampolli) [inline Detectors/GlobalTracking/include/GlobalTracking/MatchITSTPCQC.h]: what is the problem in keeping them?

Comment (f3sch) [reply Detectors/GlobalTracking/include/GlobalTracking/MatchITSTPCQC.h]: No problem at all just wondering if this is deadcode or if there is use case? Anyways I just commented them out to test if something will fail.

**Comment by chiarazampolli** [comment]
Before merging, we need to check if TH*F is enough for e.g. an 8h long run in pp and PbPb, @mfaggin , @noferini

**Comment by chiarazampolli** [comment]
Hello, 

According to https://docs.google.com/spreadsheets/d/1sXokWFLjyceNhTlTcKL_-CczFZ6Vy2IMeCUJ2ujsTWs/edit#gid=0, the gain will anyway be minimal. @f3sch , why don't we keep all TH*D, and change the rest in this PR?

Chiara

**Comment by chiarazampolli** [comment]
Hello @f3sch , 
Now you kept the TH*D, right?
@shahor02 , can you please check and approve if ok?
Chiara

**Comment by f3sch** [comment]
Hi @chiarazampolli, yes I did revert this part.

**Comment by chiarazampolli** [comment]
Thanks @f3sch !
@shahor02 , the error does not seem related to this PR, what do you think?
Chiara

**Review by shahor02 — APPROVED** [review]
Hi, looks fine, the failure is unrelated. 
Eventually, I would substitute all pT axes by 1/pT and use less bins, we still have too many 2D histos with 100x200 bins.
@chiarazampolli I did not understand your reference on https://docs.google.com/spreadsheets/d/1sXokWFLjyceNhTlTcKL_-CczFZ6Vy2IMeCUJ2ujsTWs/edit#gid=0 concerning TH*D vs TH*F effect: there should be 50% difference for TH1*, and factor 2 for TH2*, so in latter case the double version should be avoided (as it is in this code).

**Comment by chiarazampolli** [comment]
Hello @shahor02 , 

We also have QC vs 1/pt already, but not 2D.

For the TH*D vs TH*F, Felix had changed from TH*D to TH*F in this PR. We then were not sure since we might hit the limit of precision of TH*F... So we preferred to keep TH*D, considering also that the gain in memory did not look so big according to Felix's expectation (written in the spreadsheet). 

Does this clarify?

cheers,
Chiara

**Comment by shahor02** [comment]
this was actually my question, in the spreadsheet I did not see anything about THxD vs THxF...

**Pull request merged**
